autogen/oai/client.py

```diff
@@ -240,17 +242,18 @@ def yes_or_no_filter(context, response):
                             response.pass_filter = pass_filter
                             # TODO: add response.cost
                             return response
-                try:
-                    response = self._completions_create(client, params)
-                except APIError:
-                    logger.debug(f"config {i} failed", exc_info=1)
-                    if i == last:
-                        raise
-                else:
-                    if cache_seed is not None:
-                        # Cache the response
+            try:
+                response = self._completions_create(client, params)
+            except APIError:
+                logger.debug(f"config {i} failed", exc_info=1)
+                if i == last:
+                    raise
+            else:
```

Comment:
Check the filter first.

Reply:
I believe I have corrected the issue. Please have a look.

There is some question on if the result should be saved to cache if it fails a filter. I believe it should -- otherwise repeated calls are perhaps likely to also fail and thus be wasteful. On the other hand, one might argue that such responses should not be cached, since repeated calls give it an extra chance to pass.